### ISSUE TITLE
Make "repeat last insert" respect the number prefix

### ISSUE THREAD

**Issue by xJonathanLEI** [helix-editor/helix#4359]
With the current implementation, when you press `<n>.`, the last insert is repeated only once, completely ignoring the number `<n>`, unlike in vim where it's indeed inserted `<n>` times.

This feature is useful for inserting repeated characters. I currently work around this issue by recording the insert as a macro, and do `<n>q` instead, but this isn't very efficient.

I would like to implement this feature, but would love to get a green light from the core team first just in case the current behavior is actually intended. Thanks!

**Comment by Omnikar** [comment]
I'd say you can go ahead and open a PR for this; the current behaviour definitely isn't explicitly intended.

**Comment by the-mikedavis** [comment]
There's a separate issue for that: https://github.com/helix-editor/helix/issues/1488
